src/Nethermind/Nethermind.JsonRpc/Data/TransactionForRpc.cs

```diff
@@ -151,23 +128,9 @@ public Transaction ToTransaction(ulong? chainId = null)
             return tx;
         }
 
-        private AccessList? TryGetAccessList()
-        {
-            if (Type != TxType.AccessList || AccessList == null)
-            {
-                return null;
-            }
-
-            AccessListBuilder accessListBuilder = new();
-            for (int i = 0; i < AccessList.Length; i++)
-            {
-                accessListBuilder.AddAddress(AccessList[i].Address);
-                for (int j = 0; j < AccessList[i].StorageKeys.Length; j++)
-                {
-                    accessListBuilder.AddStorage(AccessList[i].StorageKeys[j]);
-                }
-            }
-            return accessListBuilder.ToAccessList();
-        }
+        private AccessList? TryGetAccessList() =>
+            Type != TxType.AccessList || AccessList == null 
```

Comment:
does it modify the actual transaction content depending on type? why would this check be required?

Reply:
To create proper object according to spec?